src/main/java/net/minecraftforge/common/ForgeSpawnEggItem.java

```diff
@@ -0,0 +1,122 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common;
+
+import net.minecraft.block.DispenserBlock;
+import net.minecraft.dispenser.DefaultDispenseItemBehavior;
+import net.minecraft.dispenser.IBlockSource;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.SpawnReason;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.SpawnEggItem;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.Direction;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.client.event.ColorHandlerEvent;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
+
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.function.Supplier;
+
+public class ForgeSpawnEggItem extends SpawnEggItem
+{
+    private static final List<ForgeSpawnEggItem> MOD_EGGS = new ArrayList<>();
+    private static final Map<EntityType<?>, ForgeSpawnEggItem> TYPE_MAP = new IdentityHashMap<>();
+    private final Supplier<? extends EntityType<?>> typeSupplier;
+
+    public ForgeSpawnEggItem(Supplier<? extends EntityType<?>> type, int backgroundColor, int highlightColor, Properties props)
+    {
+        super((EntityType<?>) null, backgroundColor, highlightColor, props);
+        this.typeSupplier = type;
+
+        MOD_EGGS.add(this);
+    }
+
+    @Override
+    public EntityType<?> getType(@Nullable CompoundNBT tag)
+    {
+        EntityType<?> type = super.getType(tag);
+        return type != null ? type : typeSupplier.get();
+    }
+
+    @Nullable
+    protected DefaultDispenseItemBehavior createDispenseBehavior()
+    {
+        return DEFAULT_DISPENSE_BEHAVIOR;
+    }
+
+    @Nullable
+    public static SpawnEggItem fromEntityType(@Nullable EntityType<?> type)
+    {
+        SpawnEggItem ret = TYPE_MAP.get(type);
+        return ret != null ? ret : SpawnEggItem.byId(type);
+    }
+
+
+
+    private static final DefaultDispenseItemBehavior DEFAULT_DISPENSE_BEHAVIOR = new DefaultDispenseItemBehavior()
+    {
+        protected ItemStack execute(IBlockSource source, ItemStack stack)
+        {
+            Direction face = source.getBlockState().getValue(DispenserBlock.FACING);
+            EntityType<?> type = ((SpawnEggItem) stack.getItem()).getType(stack.getTag());
+
+            type.spawn(source.getLevel(), stack, null, source.getPos().relative(face), SpawnReason.DISPENSER, face != Direction.UP, false);
```

Comment:
the 1.17 version has a try/catch here, is it relevant in the backport?

Reply:
The dispense behaviors are a copy of the vanilla behaviors used by the normal spawn egg and Mojang added a try/catch there in 1.17.
If you want, I can add the try/catch here anyway.